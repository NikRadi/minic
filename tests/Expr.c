int main() {
    1; 2; 3;;; 4;;;
    { 1; { 2; }}
    printf("%d\n", 0);
    printf("%d\n", 42);
    printf("%d\n", 5 + 20 - 4);
    printf("%d\n", 12 + 34 - 5);
    printf("%d\n", 5 + 6 * 7);
    printf("%d\n", 5 * (9 - 6));
    printf("%d\n", (3 + 5) / 2);
    printf("%d\n", -10 + 20);
    printf("%d\n", --10);
    printf("%d\n", --+10);
    printf("%d\n", 0 == 1);
    printf("%d\n", 0 != 1);
    printf("%d\n", 0 < 1);
    printf("%d\n", 1 < 1);
    printf("%d\n", 2 < 1);
    printf("%d\n", 0 <= 1);
    printf("%d\n", 1 <= 1);
    printf("%d\n", 2 <= 1);
    printf("%d\n", 0 > 1);
    printf("%d\n", 1 > 1);
    printf("%d\n", 2 > 1);
    printf("%d\n", 0 >= 1);
    printf("%d\n", 1 >= 1);
    printf("%d\n", 2 >= 1);
}
