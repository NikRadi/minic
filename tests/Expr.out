0
42
21
41
47
15
4
10
10
10
0
1
1
0
0
1
1
0
0
0
1
0
1
1